modules/calc/src/main/java/com/opengamma/strata/calc/runner/TaskBuilder.java

```diff
@@ -0,0 +1,204 @@
+   * 
+   * @return the calculation target
+   */
+  public CalculationTarget getTarget() {
+    return target;
+  }
+
+  /**
+   * Gets the row index.
+   * 
+   * @return the row index
+   */
+  public int getRowIndex() {
+    return rowIndex;
+  }
+
+  /**
+   * Gets the function.
+   * 
+   * @return the calculation function
+   */
+  public CalculationFunction<?> getFunction() {
+    return function;
+  }
+
+  /**
+   * Gets the columns.
+   * 
+   * @return the calculation columns
+   */
+  public List<TaskColumn> getColumns() {
+    return columns;
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Causes the specified column to be executed as a separate task.
+   * <p>
+   * This might be used if a measure is significantly slower to calculate than other measures.
+   * By executing separately, the calculation of the slow measure will not block calculation
+   * of other measures.
+   * 
+   * @param column  the column to execute separately
+   * @return this builder
+   */
+  public TaskBuilder executeSeparately(TaskColumn column) {
+    separate[column.getColumnIndex()] = true;
+    return this;
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Builds the tasks based on the state of the builder.
+   * 
+   * @return the tasks
+   */
+  public List<CalculationTask> build() {
+    ImmutableList.Builder<CalculationTask> taskBuilder = ImmutableList.builder();
+
+    // create the cells and group them
+    ListMultimap<GroupingKey, CalculationTaskCell> grouped = ArrayListMultimap.create();
+    for (int colIndex = 0; colIndex < columns.size(); colIndex++) {
+      TaskColumn column = columns.get(colIndex);
+      CalculationTaskCell cell =
+          CalculationTaskCell.of(rowIndex, colIndex, column.getMeasure(), column.getReportingCurrency());
+      if (separate[colIndex]) {
+        taskBuilder.add(CalculationTask.of(
+            target, function, column.getMarketDataMappings(), column.getParameters(), ImmutableList.of(cell)));
+      } else {
+        grouped.put(new GroupingKey(column.getMarketDataMappings(), column.getParameters()), cell);
+      }
+    }
+
+    // build tasks
+    for (GroupingKey key : grouped.keySet()) {
+      taskBuilder.add(CalculationTask.of(target, function, key.mappings, key.parameters, grouped.get(key)));
+    }
+    return taskBuilder.build();
+  }
+
+  // key used for grouping
+  static final class GroupingKey {
+    final MarketDataMappings mappings;
+    final CalculationParameters parameters;
+
+    GroupingKey(MarketDataMappings mappings, CalculationParameters parameters) {
+      this.mappings = mappings;
+      this.parameters = parameters;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      if (obj instanceof GroupingKey) {
+        GroupingKey other = (GroupingKey) obj;
+        // mappings matched by ==, parameters by equals()
+        return mappings == other.mappings &&
```

Comment:
Mappings are not guaranteed to have an equals(), and if they do it might be complex. But I'll change it as it should cause no harm.